hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/classic/ClassicOperationExecutor.java

```diff
@@ -65,58 +70,44 @@
     private final ILogger logger;
 
     // all operations for specific partitions will be executed on these threads, e.g. map.put(key, value)
-    private final PartitionOperationThread[] partitionOperationThreads;
+    private final PartitionOperationThread[] partitionThreads;
     private final OperationRunner[] partitionOperationRunners;
 
-    private final OperationQueue genericOperationQueue;
+    private final OperationQueue genericQueue = new DefaultOperationQueue();
 
     // all operations that are not specific for a partition will be executed here, e.g. heartbeat or map.size()
-    private final GenericOperationThread[] genericOperationThreads;
+    private final GenericOperationThread[] genericThreads;
     private final OperationRunner[] genericOperationRunners;
 
     private final Address thisAddress;
-    private final NodeExtension nodeExtension;
-    private final HazelcastThreadGroup threadGroup;
     private final OperationRunner adHocOperationRunner;
-    private final MetricsRegistry metricsRegistry;
 
     public ClassicOperationExecutor(GroupProperties properties,
                                     LoggingService loggerService,
                                     Address thisAddress,
                                     OperationRunnerFactory operationRunnerFactory,
-                                    HazelcastThreadGroup hazelcastThreadGroup,
+                                    HazelcastThreadGroup threadGroup,
                                     NodeExtension nodeExtension,
                                     MetricsRegistry metricsRegistry) {
         this.thisAddress = thisAddress;
-        this.nodeExtension = nodeExtension;
-        this.threadGroup = hazelcastThreadGroup;
-        this.metricsRegistry = metricsRegistry;
         this.logger = loggerService.getLogger(ClassicOperationExecutor.class);
-        this.genericOperationQueue = new DefaultOperationQueue();
 
         this.adHocOperationRunner = operationRunnerFactory.createAdHocRunner();
 
         this.partitionOperationRunners = initPartitionOperationRunners(properties, operationRunnerFactory);
-        this.partitionOperationThreads = initPartitionThreads(properties);
+        this.partitionThreads = initPartitionThreads(properties, threadGroup, nodeExtension, metricsRegistry);
 
         this.genericOperationRunners = initGenericOperationRunners(properties, operationRunnerFactory);
-        this.genericOperationThreads = initGenericThreads();
+        this.genericThreads = initGenericThreads(threadGroup, nodeExtension, metricsRegistry);
 
-        logger.info("Starting with " + genericOperationThreads.length + " generic operation threads and "
-                + partitionOperationThreads.length + " partition operation threads.");
-    }
+        metricsRegistry.scanAndRegister(this, "operation");
```

Comment:
In this case there's already the start method so it should be simple.

On Wed, Mar 30, 2016, 16:55 Donnerbart <notifications@github.com> wrote:

> In
> hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/classic/ClassicOperationExecutor.java
> <https://github.com/hazelcast/hazelcast/pull/7863#discussion_r57902592>:
>
> >
> > -        logger.info("Starting with " + genericOperationThreads.length + " generic operation threads and "
> > -                + partitionOperationThreads.length + " partition operation threads.");
> > -    }
> > +        metricsRegistry.scanAndRegister(this, "operation");
>
> Instructions can be re-ordered? This is a huge no-go in our favorite book
> from Brian Goetz (3.2.1 Safe construction practices). So it maybe wrong in
> other parts of the system as well.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/hazelcast/hazelcast/pull/7863/files/fbd2e9abc9e31c23cc704447ec564accfe5b0b69#r57902592>
>